src/cluster_legacy.c

```diff
@@ -4672,10 +4672,14 @@ void clusterCron(void) {
             /* Timeout reached. Set the node as possibly failing if it is
              * not already in this state. */
             if (!(node->flags & (CLUSTER_NODE_PFAIL|CLUSTER_NODE_FAIL))) {
-                serverLog(LL_DEBUG,"*** NODE %.40s possibly failing",
-                    node->name);
                 node->flags |= CLUSTER_NODE_PFAIL;
                 update_state = 1;
+                if (clusterNodeIsMaster(myself) && server.cluster->size == 1) {
```

Comment:
We should add a comment here for why we are doing this, the server.cluster->size check is not obvious.

Reply:
I also noticed that server.cluster->size is [initialized to 1](https://github.com/redis/redis/blob/e2a3f3091fa333b1856a0a6f2b28760f2aafbe7d/src/cluster_legacy.c#L962), even when the primary has no slots assigned. It has been observed that server.cluster->size will be reported as 1 in the current function for up to two seconds before being updated to zero [here](https://github.com/redis/redis/blob/e2a3f3091fa333b1856a0a6f2b28760f2aafbe7d/src/cluster_legacy.c#L4908). Do you think this introduces the possibility that a primary with zero slots can mark a replica as failed or is the time window too small to worry about?

Reply:
I think that makes sense to start at zero. Let's make that change here too.